vcd/resource_vcd_nsxt_alb_service_engine_group.go

```diff
@@ -0,0 +1,249 @@
+package vcd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceVcdAlbServiceEngineGroup() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVcdAlbServiceEngineGroupCreate,
+		ReadContext:   resourceVcdAlbServiceEngineGroupRead,
+		UpdateContext: resourceVcdAlbServiceEngineGroupUpdate,
+		DeleteContext: resourceVcdAlbServiceEngineGroupDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdAlbServiceEngineGroupImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "NSX-T ALB Service Engine Group name",
+			},
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "NSX-T ALB Service Engine Group description",
+			},
+			"alb_cloud_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "NSX-T ALB backing Cloud ID",
+			},
+			"reservation_model": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "NSX-T ALB Service Engine Group reservation model. One of 'DEDICATED', 'SHARED'",
+				ValidateFunc: validation.StringInSlice([]string{"DEDICATED", "SHARED"}, false),
+			},
+			// Ideally this should be a reference by ID and a data source for lookup. However, the Importable Service
+			// Engine Group API endpoint does not return entities once they are consumed, and it is impossible to make
+			// a data source.
+			"service_engine_group_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "NSX-T ALB Importable Service Engine Group Name",
```

Comment:
Just a point for thought: where do we draw the line of **not** using `_importable_` in the name? It's OK not to use it here, we just need to be consistent with the approach :)

Reply:
I have added this longer name to correctly reflect it.